design/003-interactive-session/003-interactive-session.md

```diff
@@ -0,0 +1,206 @@
+
+##### Detailed Parameter Description
+
+```
+renku session start [OPTIONS]
+
+--image <image_name>			Override the docker image to be used for the interactive session.
+-p|--provider [docker|renkulab] 	Use the specified backend for starting an interactive session.
+-c|--config   				YAML file containing configuration for the provider.
+--cpu <num of cpus>			Specify how much of the available CPU resources a container can use.
+--disk <disk space>			Specify the amount of disk space for the container.
+--gpu <device>				Specify the GPU allocated for the container.
+--memory <max memory>			The maximum amount of memory the container can use.
+
+```
+
+#### renku session stop
+
+`renku session stop my_session` simply stops the given interactive session.
+
+One can shut down all the currently running interactive sessions at once with the `renku session stop --all`.
+
+##### Detailed Parameter Description
+
+```
+renku session stop [OPTIONS] <name>
+
+<name>      				Name of the session that the user wants to stop.
+
+--all       				Stops all the running containers.
+-p|--provider [docker|renkulab] 	Use the specified backend for starting an interactive session.
+
+```
+
+#### renku session list
+
+The command to list all the currently running interactive sessions for the given project, with the
+detailed information regarding the network port mappings.
+
+##### Detailed Parameter Description
+
+```
+renku session list [OPTIONS]
+
+-p|--provider [docker|renkulab] 	Use the specified backend for listing the interactive sessions.
+-c|--config   				YAML file containing configuration for the provider.
+
+```
+
+#### renku session open
+
+`renku session open <my_session>` opens the given interactive session in the user's default browser.
+
+Note, it is possible that the user is running renku on a remote machine via SSH. In this case this command
+might fail, unless for example X-Fowarding is not enabled.
+
+##### Detailed Parameter Description
+
+```
+renku session open <name>
+
+<name>      				Name of the session that the user wants to stop.
+
+-p|--provider [docker|renkulab] 	Use the specified backend for listing the interactive sessions.
+
+```
+
+## Drawbacks
+
+One of the major drawback of the current design is the dependency on docker. Namely, to only support local
+interactive session if docker is available. Many HPC environments do not support docker, but they support
+other container formats, like [runC](https://github.com/opencontainers/runc). As a long term goal of this
+effort, support for different container engines is essential. In order to allow extending the `renku session`
+command with different container engines a very simple API is defined.
+
+The interface requires that the container engine support the following functionalities:
+ - finding a specific container image. Both locally (cached) and as well in a remote repository.
```

Comment:
we are not dealing with authentication. this is a wrapper around working with the containers... the authentication shall be done from the container manager side.

Reply:
if the user has previously done a `docker login` then the pull will be able to pull from a private repo... if that's not done it won't be able and then it will complain that it cannot find the image.

Reply:
yeah but find image will fail if you try to look at a private registry and don't authenticate. So you're just saying the user will have to do that by some other means?

Reply:
ok so in our case maybe `renku login` needs to handle the authentication with the gitlab registry...

Reply:
no what i mean is that if you try that you'll see that you'll get an error from docker that says try to use `docker login` coz it couldn't find on the public repos the image that you are looking for

Reply:
of course, but our users won't really know what to do with that probably. That's why I thought login could be one of the parts of the interface.